File: app/widgets/molecular/primers/purification_method_auto_complete_field.rb
class Molecular::Primers::PurificationMethodAutoCompleteField < Templates::AutoCompleteField
  def initialize options
    @primer ||= options[:primer] || fail('primer not given')
    @model_object ||= @primer
    @attribute_path ||= 'purification_method'
    @value_method = 'id'
    @text_method = 'name'
    @width = 275
    super
  end
end

File: app/widgets/molecular/primers/purification_method_auto_complete_field.js
//= require <templates/auto_complete_field>
//= require <molecular/purification_method>

JooseModule('Molecular.Primers', function () {
  JooseClass('PurificationMethodAutoCompleteField', {
    isa: Templates.AutoCompleteField,
    has: {
      primer: { is: 'ro', required: true, nullable: false },
      object: { is: 'ro', init: function () { return this.primer() } },
      method:        { is: 'ro', init: 'purification_method' },
      valueMethod:   { is: 'ro', init: 'id' },
      textMethod:    { is: 'ro', init: 'name' },
      width:         { is: 'ro', init: 275 },
      collection: { is: 'ro', init: function () { return (
        Molecular.PurificationMethod.
          collection({ context: this.context() }).
          select('id', 'name')
      ) } }
    }
  })
})
